libfwupd/fwupd-enums-private.h

```diff
@@ -10,61 +10,500 @@
-#define FWUPD_RESULT_KEY_PLUGIN			"Plugin"	/* s */
-#define FWUPD_RESULT_KEY_PROTOCOL		"Protocol"	/* s */
-#define FWUPD_RESULT_KEY_RELEASE		"Release"	/* a{sv} */
-#define FWUPD_RESULT_KEY_REMOTE_ID		"RemoteId"	/* s */
-#define FWUPD_RESULT_KEY_SERIAL			"Serial"	/* s */
-#define FWUPD_RESULT_KEY_SIZE			"Size"		/* t */
-#define FWUPD_RESULT_KEY_STATUS			"Status"	/* u */
-#define FWUPD_RESULT_KEY_SUMMARY		"Summary"	/* s */
-#define FWUPD_RESULT_KEY_BRANCH			"Branch"	/* s */
-#define FWUPD_RESULT_KEY_TRUST_FLAGS		"TrustFlags"	/* t */
-#define FWUPD_RESULT_KEY_UPDATE_MESSAGE		"UpdateMessage"	/* s */
-#define FWUPD_RESULT_KEY_UPDATE_IMAGE		"UpdateImage"	/* s */
-#define FWUPD_RESULT_KEY_UPDATE_ERROR		"UpdateError"	/* s */
-#define FWUPD_RESULT_KEY_UPDATE_STATE		"UpdateState"	/* u */
-#define FWUPD_RESULT_KEY_URI			"Uri"		/* s */
-#define FWUPD_RESULT_KEY_LOCATIONS		"Locations"	/* as */
-#define FWUPD_RESULT_KEY_VENDOR_ID		"VendorId"	/* s */
-#define FWUPD_RESULT_KEY_VENDOR			"Vendor"	/* s */
-#define FWUPD_RESULT_KEY_VENDOR			"Vendor"	/* s */
-#define FWUPD_RESULT_KEY_VERSION_BOOTLOADER	"VersionBootloader"	/* s */
-#define FWUPD_RESULT_KEY_VERSION_BOOTLOADER_RAW	"VersionBootloaderRaw"	/* t */
-#define FWUPD_RESULT_KEY_VERSION_FORMAT		"VersionFormat"	/* u */
-#define FWUPD_RESULT_KEY_VERSION_RAW		"VersionRaw"	/* t */
-#define FWUPD_RESULT_KEY_VERSION_LOWEST		"VersionLowest"	/* s */
-#define FWUPD_RESULT_KEY_VERSION_LOWEST_RAW	"VersionLowestRaw"	/* t */
-#define FWUPD_RESULT_KEY_VERSION		"Version"	/* s */
+/**
+ * FWUPD_RESULT_KEY_APPSTREAM_ID:
+ *
+ * Result key to represent AppstreamId
+ *
+ * Type of 's'
+ *
+ **/
+#define FWUPD_RESULT_KEY_APPSTREAM_ID		"AppstreamId"
+/**
+ * FWUPD_RESULT_KEY_CHECKSUM:
+ *
+ * Result key to represent Checksum
+ *
+ * Type of 'as'
+ *
+ **/
+#define FWUPD_RESULT_KEY_CHECKSUM		"Checksum"
+/**
+ * FWUPD_RESULT_KEY_CREATED:
+ *
+ * Result key to represent Created
+ *
+ * Type of 't'
+ *
+ **/
+#define FWUPD_RESULT_KEY_CREATED		"Created"
+/**
+ * FWUPD_RESULT_KEY_DESCRIPTION:
+ *
+ * Result key to represent Description
+ *
+ * Type of 's'
+ *
+ **/
+#define FWUPD_RESULT_KEY_DESCRIPTION		"Description"
+/**
+ * FWUPD_RESULT_KEY_DETACH_CAPTION:
+ *
+ * Result key to represent DetachCaption
+ *
+ * Type of 's'
+ *
+ **/
+#define FWUPD_RESULT_KEY_DETACH_CAPTION		"DetachCaption"
+/**
+ * FWUPD_RESULT_KEY_DETACH_IMAGE:
+ *
+ * Result key to represent DetachImage
+ *
+ * Type of 's'
+ *
+ **/
+#define FWUPD_RESULT_KEY_DETACH_IMAGE		"DetachImage"
+/**
+ * FWUPD_RESULT_KEY_DEVICE_ID:
+ *
+ * Result key to represent DeviceId
+ *
+ * Type of 's'
+ *
+ **/
+#define FWUPD_RESULT_KEY_DEVICE_ID		"DeviceId"
+/**
+ * FWUPD_RESULT_KEY_PARENT_DEVICE_ID:
+ *
+ * Result key to represent ParentDeviceId
+ *
+ * Type of 's'
+ *
+ **/
+#define FWUPD_RESULT_KEY_PARENT_DEVICE_ID	"ParentDeviceId"
+/**
+ * FWUPD_RESULT_KEY_COMPOSITE_ID:
+ *
+ * Result key to represent CompositeId
```

Comment:
I think we should probably mention things like fwupd_device_get_composite_id() here too.

Reply:
On the result key?  That's going to be a lot of work in what I think is little value.  It's going to start precedent of trying to remember everywhere that a data type if used.

In this I think it would be _better_ to modify `fwupd_device_get_composite_id` (and similar functions) to point to their types in the constants when used.  This is of course also a lot of work, but I think more useful work.

Reply:
Yes, agreed.